src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -134,16 +134,21 @@ public Uri CommitFile(string fileName)
                 {
                     uri = CommitFileRetry(fileName);
                 }
-                catch (RequestFailedException)
+                catch (RequestFailedException ex)
                 {
+                    _logger.LogError(ex, "Failed to write export file");
                     try
                     {
                         uri = CommitFileRetry(fileName);
                     }
-                    catch (RequestFailedException ex)
+                    catch (ObjectDisposedException odEx)
                     {
-                        _logger.LogError(ex, "Failed to write export file");
-                        throw new DestinationConnectionException(ex.Message, (HttpStatusCode)ex.Status);
+                        _logger.LogError(odEx, "Failed to write export file due to ObjectDisposedException");
```

Comment:
I think we need a similar throw of `DestinationConnectionException` with ex here. Without throwing it's possible we report an export as successful when all the data wasn't written.